### PR TITLE
Create local smoketest

### LINKED ISSUE #491: Reenable smoketests
In https://github.com/raiden-network/scenario-player/pull/489 it showed that the smoketest currently fail: https://app.circleci.com/jobs/github/raiden-network/scenario-player/3625

I disabled them to be able to land the PR quickly, but we need to re-enable the smoketests.

give a try to do the proper fix according to @konradkonrad like the smoketest in Raiden

timeboxed to 2 days

### PR DESCRIPTION
**Pull request by konradkonrad**
Implement a `scenario_player smoketest` command.

This reuses  https://github.com/raiden-network/raiden/pull/5966 in order to start a local smoke testing scenario player run.

The smoketest is independent of any networked resources (i.e. blockchain, transport, services).

Related changes included:
- Make `--chain` a cli only setting (don't read the setting from `.yaml` anymore)
- Removed a global gas estimation monkeypatch. IF the monkeypatching is still required, it should be applied selectively and sensible to the EthClient in use.

~~Depends on https://github.com/raiden-network/raiden/pull/5985~~ (done)

Fixes #491

### CODE REVIEW

File: tests/unittests/cli/test_cli.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 CLI_ARGS = (
     f"--keystore-file {KEYSTORE_PATH.joinpath('UTC--1')} "
     f"--no-ui "
+    "--chain smoketest:http://localhost:12345 "
```
Comment (fredo) [inline tests/unittests/cli/test_cli.py]: is that necessary here? Or rather whats the purpose?

File: tests/unittests/cli/test_cli.py [hunk 2]
```diff
@@ -118,6 +119,7 @@ def test_reclaim_eth_data_path(self, runner, tmpdir):
         result = runner.invoke(
             main.reclaim_eth,
             f"--data-path {path_arg} "
+            "--chain smoketest:http://localhost:12345 "
```
Comment (fredo) [inline tests/unittests/cli/test_cli.py]: same here

File: pyproject.toml [hunk 3]
```diff
@@ -59,6 +59,7 @@ keywords = "raiden, raiden-network"
 # these automatically.
 test = ["pytest==4.5.0"]
 dev = [
+    "matrix-synapse==1.5.1",
```
[the remark this answers is not in the document]
Comment (konradkonrad) [reply pyproject.toml]: done

File: tests/unittests/cli/test_cli.py (hunk 1 above)
Comment (konradkonrad) [reply tests/unittests/cli/test_cli.py]: https://github.com/raiden-network/scenario-player/pull/516/commits/beead0c0cc95518380f9014adffde95cdde682f3 makes the `--chain ...` argument mandatroy, so the cli tests don't run without it anymore

File: pyproject.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (konradkonrad) [reply pyproject.toml]: erm...looking again at this. at least at the moment, `raiden-network/raiden` still does depend on 1.5.1: https://github.com/raiden-network/raiden/blob/99ec6c606d98f8e672deee7c4721b8f820e41382/requirements/requirements-dev.txt#L86

[the remark this answers is not in the document]
Comment (fredo) [reply pyproject.toml]: I just saw that the corresponding PR isnt merged yet

**Pull request merged**